internal/dao/stubs/pubkey_dao.go

```diff
@@ -0,0 +1,54 @@
+package stubs
+
+import (
+	"context"
+
+	"github.com/RHEnVision/provisioning-backend/internal/dao"
+	"github.com/RHEnVision/provisioning-backend/internal/models"
+)
+
+type pubkeyCtxKeyType string
+
+var pubkeyCtxKey pubkeyCtxKeyType = "pubkey-dao"
+
+type PubkeyDaoStub struct {
+	store []*models.Pubkey
+}
+
+func init() {
+	dao.GetPubkeyDao = getPubkeyDao
+}
+
+type ContextReadError struct{}
+
+func (m *ContextReadError) Error() string {
+	return "failed to find or convert dao stored in testing context"
+}
+
+func WithPubkeyDao(parent context.Context, init_store []*models.Pubkey) context.Context {
+	ctx := context.WithValue(parent, pubkeyCtxKey, PubkeyDaoStub{init_store})
+	return ctx
+}
+
+func getPubkeyDao(ctx context.Context) (dao.PubkeyDao, error) {
+	var err error
+	pkdao, ok := ctx.Value(pubkeyCtxKey).(dao.PubkeyDao)
+	if !ok {
+		err = &ContextReadError{}
```

Comment:
I think you could simply use `fmt.Errorf` here, not big deal for mocks. Please refactor this later, the ContextReadError looks like a common error that should not live in pubkey_dao.go.

Reply:
Linter wanted this, but I will try to improve so it is easier to read.